src/Storages/System/StorageSystemDictionaries.cpp

```diff
@@ -20,6 +20,72 @@
 
 namespace DB
 {
+namespace
+{
+    /// Used for column indexes!  Don't change the numbers!
+    enum SystemDictionariesColumn
+    {
+        Database = 0,
+        Name,
+        Uuid,
+        Status,
+        Origin,
+        Type,
+        KeyNames,
+        KeyTypes,
+        AttributeNames,
+        AttributeTypes,
+        BytesAllocated,
+        HierarchicalIndexBytesAllocated,
+        QueryCount,
+        HitRate,
+        FoundRate,
+        ElementCount,
+        LoadFactor,
+        Source,
+        LifetimeMin,
+        LifetimeMax,
+        LoadingStartTime,
+        LastSuccessfulUpdateTime,
+        LoadingDuration,
+        LastException,
+        Comment,
+        /// Virtual columns
+        Key
+    };
+
+    std::shared_ptr<const IDictionary> getDictionaryPtr(const ExternalLoader::LoadResult & load_result)
```

Comment:
Consider to remove this method, it just implements `dynamic_pointer_cast`.

Reply:
Removed